pyro/infer/mcmc/mcmc.py

```diff
@@ -49,7 +52,12 @@ def _traces(self, *args, **kwargs):
                     self.logger.info(diagnostic_info)
             trace = self.kernel.sample(trace)
             if self._t < self.warmup_steps:
```

Comment:
Can we simplify this logic as follows:
```
self._t += 1
if self._t <= self.warmup_steps or self._t % self.lag != 0:
  continue
yield (trace, Variable(torch.Tensor([1.0])))
```

We should also change the while loop above to `<=`.